lib/utils/reflection.utilts.ts

```diff
@@ -100,14 +100,14 @@ function createWrappedExplicitTypeFn(
   explicitTypeFn: ReturnTypeFunc,
   options: TypeOptions,
 ) {
-  return () => {
-    const explicitTypeRef = explicitTypeFn();
-    if (Array.isArray(explicitTypeRef)) {
-      const { depth, typeRef } = getTypeReferenceAndArrayDepth(explicitTypeRef);
-      options.isArray = true;
-      options.arrayDepth = depth;
-      return typeRef;
-    }
-    return explicitTypeRef;
-  };
+  let explicitTypeRef = explicitTypeFn();
```

Comment:
As far as I remember, this line must be called inside `() =>` (lazily) to prevent circular-dependency-related issues.

Reply:
Can you provide me a case when it can cause the error?

I just tried this, and it works with this code. Or this is not the circular-dependency?

```js
  @ObjectType()
  class ClassA {
    @Field({ nullable: true })
    login: string;

    @Field()
    password: string;
  }

  @ObjectType()
  class ClassB {
    @Field()
    lastName: string;

    firstName?: string;

    @Field(() => [String])
    hobbies?: string[];

    @Field(() => [[ClassB]])
    matrix?: [ClassB[]];

    static [METADATA_FACTORY_NAME]() {
      return {
        firstName: { nullable: true, type: () => String },
      };
    }
  }
```

Btw I am try to rewrite this.

Reply:
Okay, then the solution would be to put the options variable in callback like get typeFn works. In this callback I can calculate options, like array depth etc. But as I see this will require a lot of rework. Can you advise better solution? Or I can start to do my variant.

Reply:
@kamilmysliwiec ?

Reply:
@kamilmysliwiec  I really need this fix ASAP, can you answer me?

Reply:
@kamilmysliwiec ?

Reply:
@kamilmysliwiec ?

Reply:
@kamilmysliwiec so, are you interested to fix this bug?